From andrew@lunn.ch  Tue Jan  5 14:39:37 2010
From: Sven Eckelmann <sven.eckelmann@gmx.de>
Date: Sat,  2 Jan 2010 11:30:36 +0100
Subject: Staging: batman-adv: Remove NULL pointer check
To: gregkh@suse.de
Cc: b.a.t.m.a.n@lists.open-mesh.net, Sven Eckelmann <sven.eckelmann@gmx.de>, Andrew Lunn <andrew@lunn.ch>
Message-ID: <1262428252-26439-1-git-send-email-andrew@lunn.ch>


From: Sven Eckelmann <sven.eckelmann@gmx.de>

It is safe to call kfree(NULL) which makes this extra check unneeded. It
was found using checkpatch.pl from linux-2.6

Signed-off-by: Sven Eckelmann <sven.eckelmann@gmx.de>
Acked-by: Simon Wunderlich <siwu@hrz.tu-chemnitz.de>
Signed-off-by: Andrew Lunn <andrew@lunn.ch>
Signed-off-by: Greg Kroah-Hartman <gregkh@suse.de>

---
 drivers/staging/batman-adv/hard-interface.c |    3 +--
 1 file changed, 1 insertion(+), 2 deletions(-)

--- a/drivers/staging/batman-adv/hard-interface.c
+++ b/drivers/staging/batman-adv/hard-interface.c
@@ -399,8 +399,7 @@ int hardif_add_interface(char *dev, int 
 	return 1;
 
 out:
-	if (batman_if->packet_buff)
-		kfree(batman_if->packet_buff);
+	kfree(batman_if->packet_buff);
 	kfree(batman_if);
 	kfree(dev);
 	return -1;
